ledger/participant-integration-api/src/main/scala/platform/store/dao/JdbcLedgerDao.scala

```diff
@@ -456,37 +488,55 @@ private class JdbcLedgerDao(
       offsetStep: OffsetStep,
       transaction: CommittedTransaction,
       divulged: Iterable[DivulgedContract],
-      blindingInfo: Option[BlindingInfo],
   )(implicit loggingContext: LoggingContext): Future[PersistenceResponse] =
     dbDispatcher
       .executeSql(metrics.daml.index.db.storeTransactionDbMetrics) { implicit conn =>
-        val error =
-          Timed.value(
-            metrics.daml.index.db.storeTransactionDbMetrics.commitValidation,
-            postCommitValidation.validate(
-              transaction = transaction,
-              transactionLedgerEffectiveTime = ledgerEffectiveTime,
-              divulged = divulged.iterator.map(_.contractId).toSet,
-            ),
-          )
-        if (error.isEmpty) {
-          preparedInsert.write(metrics)
-          Timed.value(
-            metrics.daml.index.db.storeTransactionDbMetrics.insertCompletion,
-            submitterInfo
-              .map(prepareCompletionInsert(_, offsetStep.offset, transactionId, recordTime))
-              .foreach(_.execute()),
-          )
-        } else {
-          submitterInfo.foreach(handleError(offsetStep.offset, _, recordTime, error.get))
+        validate(ledgerEffectiveTime, transaction, divulged) match {
+          case None =>
+            preparedInsert.writeState(metrics)
+            preparedInsert.writeEvents(metrics)
+            insertCompletions(submitterInfo, transactionId, recordTime, offsetStep)
+          case Some(error) =>
+            submitterInfo.foreach(handleError(offsetStep.offset, _, recordTime, error))
         }
-        Timed.value(
-          metrics.daml.index.db.storeTransactionDbMetrics.updateLedgerEnd,
-          ParametersTable.updateLedgerEnd(offsetStep),
-        )
+
+        updateLedgerEnd(offsetStep)
         Ok
       }
 
+  private def validate(
+      ledgerEffectiveTime: Instant,
+      transaction: CommittedTransaction,
+      divulged: Iterable[DivulgedContract],
+  )(implicit connection: Connection) =
+    Timed.value(
+      metrics.daml.index.db.storeTransactionDbMetrics.commitValidation,
+      postCommitValidation.validate(
+        transaction = transaction,
+        transactionLedgerEffectiveTime = ledgerEffectiveTime,
+        divulged = divulged.iterator.map(_.contractId).toSet,
+      ),
+    )
+
+  private def insertCompletions(
+      submitterInfo: Option[SubmitterInfo],
+      transactionId: TransactionId,
+      recordTime: Instant,
+      offsetStep: OffsetStep,
+  )(implicit connection: Connection): Unit =
+    Timed.value(
+      metrics.daml.index.db.storeTransactionDbMetrics.insertCompletion,
+      submitterInfo
+        .map(prepareCompletionInsert(_, offsetStep.offset, transactionId, recordTime))
+        .foreach(_.execute()),
+    )
+
+  private def updateLedgerEnd(offsetStep: OffsetStep)(implicit connection: Connection): Unit =
```

Comment:
FYI: At one point, we had the convention that methods that take an `implicit connection: Connection` use the `Sync` suffix to make it easier to see that they don't open an extra database connection. I see only `stopDeduplicatingCommandSync` survived the many refactorings, so there's no point keeping the convention.